dsl_parser/tests/test_plugin_properties.py

```diff
@@ -0,0 +1,185 @@
+from dsl_parser.tests.abstract_test_parser import AbstractTestParser
+
+PLUGINS_YAML = """
+plugins:
+  dummy:
+    executor: central_deployment_agent
+    package_name: testing
+    source: dummy
+    properties_description: |
+      Description regarding the credentials and
+      the link to AWS documentation how to generate
+      or the link to cloudify documentation on
+      different types of authentication methods
+    properties:
+      aws_access_key_id:
+        type: string
+        description: This is a AWS Access Key ID
+        display_label: AWS Access Key ID
+      aws_secret_access_key:
+        type: string
+        description: This is a AWS Secret Access Key
+        display_label: AWS Secret Access Key
+      aws_region:
+        type: string
+        display_label: AWS Region
+"""
+
+NODES_YAML = """
+node_types:
+  type: {}
+
+node_templates:
+  dummy_node:
+    type: type
+    interfaces:
+      dummy_install:
+        install: dummy.install
+"""
+
+
+class PluginPropertiesTest(AbstractTestParser):
+    def test_parse_1_4(self):
+        with pytest.raises(DSLParsingLogicException,
+                           match='not supported in version cloudify_dsl_1_4'):
+            self.parse_1_4(PLUGINS_YAML + NODES_YAML)
+
+    def test_parse_1_5(self):
+        plan = self.parse_1_5(PLUGINS_YAML + NODES_YAML)
+        dummy = plan['deployment_plugins_to_install'][0]
+        assert dummy['properties_description'].rstrip()\
+            .endswith('authentication methods')
+        assert all(p['type'] == 'string' for p in dummy['properties'].values())
+
+    def test_no_properties(self):
+        yaml = """
+plugins:
+  dummy:
+    executor: central_deployment_agent
+    package_name: testing
+    source: dummy
+    properties_description: Empty credentials
+
+node_types:
+  type: {}
+
+node_templates:
+  dummy_node:
+    type: type
+    interfaces:
+      dummy_install:
+        install: dummy.install
+"""
+        plan = self.parse_1_5(yaml)
+        assert plan['deployment_plugins_to_install'][0]['properties'] == {}
+
+
+class PluginPropertiesImporterTest(AbstractTestParser):
+
+    def test_no_properties(self):
+        yaml = """
+imports:
+  - cloudify/types/types.yaml
+  - plugin:dummy
+"""
+        with patch('dsl_parser.elements.imports._build_ordered_imports',
+                   return_value=({}, set(), {})) as p:
+            self.parse_1_4(yaml)
+        assert p.call_count == 1
+
+    def test_basic(self):
+        yaml = """
+imports:
+  - plugin:dummy:
+      prop1: foo
+      prop2: bar
+"""
+        with patch('dsl_parser.elements.imports._build_ordered_imports',
+                   return_value=({}, set(), {})) as p:
+            self.parse_1_5(yaml)
+        assert p.call_count == 1
+
+    def test_valid(self):
+        yaml = """
+imports:
+  - cloudify/types/types.yaml
+  - plugin:dummy:
+      aws_access_key_id: foobar
+      aws_secret_access_key: {get_secret: my_secret}
```

Comment:
Shouldn't these tests also be asserting on the properties they see for the plugin? I'm not currently seeing any where we assert that we get the properties we expect (unless those properties are an empty dict)?

Reply:
The other test https://github.com/cloudify-cosmo/cloudify-common/blob/447564e2385a8a382f5ee76504d835a577095a29/dsl_parser/tests/test_plugin_properties.py#L52-L57 does that.  This one is basically about the syntax,

Reply:
That's only asserting that they're strings. It's not asserting that they have the actual content they're assigned.

Reply:
no, it asserts that all properties are of type `string` (which is most interesting), but sure, I could add a few more asserts